debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9ConstantPoolCommand.java

```diff
@@ -0,0 +1,507 @@
+                    J9ROMNameAndSignaturePointer nameAndSignature = romMethodRef.nameAndSignature();
+                    romInfo = romMethodRef.getHexValue();
+                    J9ROMClassRefPointer romClassRef = J9ROMClassRefPointer.cast(romConstantPool.add(romMethodRef.classRefCPIndex()));
+                    value = String.format("%s.%s:%s #%s",
+                                            J9UTF8Helper.stringValue(romClassRef.name()),
+                                            J9UTF8Helper.stringValue(nameAndSignature.name()),
+                                            J9UTF8Helper.stringValue(nameAndSignature.signature()),
+                                            refIndex);
+                } else if (cpType == J9ConstantPool.J9CPTYPE_ANNOTATION_UTF8) {
+                    romInfo = "!j9utf8 " + J9ROMStringRefPointer.cast(romCPP).utf8Data().getHexAddress();
+                    if (i < ramCPCount) {
+                        J9ObjectPointer stringObject = J9RAMStringRefPointer.cast(ramCPP).stringObject();
+                        if (stringObject.notNull()) {
+                            ramInfo = "!j9object " + stringObject.getHexAddress();
+                        } else {
+                            ramInfo = "(unresolved)";
+                        }
+                    }
+                    value = J9UTF8Helper.stringValue(J9ROMStringRefPointer.cast(romCPP).utf8Data());
+                } else if (cpType == J9ConstantPool.J9CPTYPE_CONSTANT_DYNAMIC) {
+                    if (i < ramCPCount) {
+                        J9RAMConstantDynamicRefPointer ref = J9RAMConstantDynamicRefPointer.cast(ramCPP);
+                        if (ref.value().notNull()) { // resolved
+                            ramInfo = "!j9object " + ref.value().getHexAddress() + " (resolved)";
+                        } else {
+                            J9ObjectPointer exception = ref.exception();
+                            if (exception.notNull()) {
+                                J9ClassPointer clazz = J9ObjectHelper.clazz(exception);
+                                J9ClassPointer throwable = J9VMConstantPoolClassRefAt(vm, J9VMCONSTANTPOOL_JAVALANGTHROWABLE).value();
+                                if (exception.getAddress() == vm.voidReflectClass().classObject().getAddress()) {
+                                    ramInfo = "NULL (resolved)";
+                                } else if (J9ClassHelper.isSameOrSuperClassOf(throwable, clazz)) { // resolved exception
+                                    ramInfo = "!j9object " + exception.getHexAddress() + " (exception)";
+                                } else { // resolving by a thread object
+                                    ramInfo = "!j9object " + exception.getHexAddress() + " (resolving Thread)";
+                                }
+                            } else {
+                                ramInfo = "(unresolved)";
+                            }
+                        }
```

Comment:
The type `J9RAMConstantDynamicRefPointer` and its fields may be unknown to the core file being examined.
We need to catch `NoClassDefFoundError` and `NoSuchFieldException` here (after updating `AuxFieldInfo29.dat` accordingly).

Reply:
Instead of
```
J9RAMConstantDynamicRef.value = required
J9RAMConstantDynamicRef.exception = required
```
it needs to say
```
J9RAMConstantDynamicRef.exception = j9object_t
J9RAMConstantDynamicRef.value = j9object_t
```
The methods generated for `exception()` and `value()` will declare that they throw `NoSuchFieldException` which must be caught. In the context of considering an entry of type `J9CPTYPE_CONSTANT_DYNAMIC`, a reasonable response is to throw `CorruptDataException`. In addition in this case, the type `J9RAMConstantDynamicRefPointer` may not be available and so `NoClassDefFoundError` must also be handled.

Reply:
I talked to Jack yesterday and he suggested casting the entry to a generic `J9RAMConstantPoolItem` and then cast the slots to the corresponding type to avoid the trouble with `J9RAMConstantDynamicRefPointer`.